modules/oadp-installing-oadp-rosa-sts.adoc

```diff
@@ -0,0 +1,123 @@
+// Module included in the following assemblies:
+//
+// * backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.adoc
+
+:_content-type: PROCEDURE
+[id="oadp-installing-oadp-rosa-sts_{context}"]
+= OADP on ROSA with STS
+
+THIS IS A DRAFT. It includes questions for Dev and QE in brackets. This draft is based on link:https://hackmd.io/yiZBeGFkSJOGIwni9iNnsA[Documentation for OADP on ROSA with STS <draft>].
+
+
+You can employ OpenShift API for Data Protection (OADP) with Red Hat OpenShift Service on AWS (ROSA) clusters to backup and restore application data. A ROSA deployment of OpenShift is configured specifically for AWS services. AWS STS is a global web service that provides short-term credentials for IAM or federated users. ROSA with STS is the recommended credential mode for ROSA clusters.
+
+[The following sentence was in the source. Is it OK? is it too vague?] Not all of the features and options provided by OADP are recommended or supported while using ROSA clusters with OADP.
+
+[IMPORTANT]
+====
+Disable Restic for OADP in ROSA environments because Restic is not supported in such environments. Use CSI storage instead.
+====
+
+.Prerequisites
+
+* A ROSA OpenShift Cluster with the required access and tokens.
+* An OpenShift secret that has been created from with your AWS token file.
++
+[QUESTION: Should I refer them to link:https://access.redhat.com/documentation/en-us/openshift_container_platform/4.10/html/backup_and_restore/application-backup-and-restore#oadp-creating-default-secret_installing-oadp-aws[Creating a default Secret] ?
```

Comment:
I don't really get what is this default secret means here: https://access.redhat.com/documentation/en-us/openshift_container_platform/4.10/html/backup_and_restore/application-backup-and-restore#oadp-creating-default-secret_installing-oadp-aws

Does it mean a secret with a default name? e.g, the default name for AWS would be `cloud-credentials`, but each provider has it's own default name. Is that what it means? If so, I think we need to elaborate that in that reference, cause it's a bit vague

Anyway, I'm not sure if in this case we need to use the default name, as there is a credentialsFile field in the VSL, so theortically could be the path of a custom name too, but I'm not sure.
@dymurray @kaovilai could you please confirm?

Reply:
default secret refers to secret with a default name which is used by VSL because it lacks ability to specify secret name as of OADP 1.1

`credentialsFile` is not intended to be used by end users, only by velero internals.

There will be an option in OADP 1.2 to specify the credentials used for VSL via a field `vsl.spec.credential` from a new enhancement by @sseago upstream https://github.com/vmware-tanzu/velero/pull/4864/files#diff-9c33e79a81a315828980d140897f96dbd273f4a83a2bc16f976658657b10d50fR35 however it is not relevant for this PR.

Reply:
@kaovilai -- should we refer them to  https://access.redhat.com/documentation/en-us/openshift_container_platform/4.10/html/backup_and_restore/application-backup-and-restore#oadp-creating-default-secret_installing-oadp-aws[Creating a default Secret] in the prerequisites or not?

Reply:
I think VSL info can be omitted. We will need to verify if VSL works first with STS before we include it IMO.

Reply:
Reading https://mobb.ninja/docs/misc/oadp/rosa-sts/ again I think we should leave the default secret prerequisites in.

Their use of credentialsFile in volumeSnapshots.velero.config is hacky and could work with some experimenting but it will still be best to use default secret name in prerequisites and wait for OADP 1.2 to land to lessen confusion with configuring VSL secret names..

Reply:
@kaovilai @RichardHoch any conclusions?